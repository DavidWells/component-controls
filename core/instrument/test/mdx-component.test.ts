import { loadStoriesTests } from './loadTestFiles';

describe('mdx-component', () => {
  loadStoriesTests(undefined, 'mdx', 'component');
});
