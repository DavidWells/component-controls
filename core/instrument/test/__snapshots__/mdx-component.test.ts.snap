// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdx-component component-parameters.mdx 1`] = `
Object {
  "components": Object {
    "2f4bd34d15be236674abaf6e894afe6e": Object {
      "from": "../../components/button-named-arrow-func",
      "importedName": "Button",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "Button",
      "request": "/Users/atanasster/component-controls/core/instrument/test/fixtures/components/button-named-arrow-func.js",
    },
  },
  "doc": Object {
    "component": "Button",
    "components": Object {
      "Button": "2f4bd34d15be236674abaf6e894afe6e",
    },
    "title": "Storybook/MDX",
  },
  "packages": Object {},
  "stories": Object {},
  "transformed": "
    
    import { Meta } from '@component-controls/blocks';
import { Button } from '../../components/button-named-arrow-func';

<Meta title=\\"Storybook/MDX\\" component={Button} />

",
}
`;

exports[`mdx-component named-alias-import.mdx 1`] = `
Object {
  "components": Object {
    "dc86c4457416c54c13147e780d972747": Object {
      "from": "../../components/button-named-class",
      "importedName": "Btn",
      "imports": undefined,
      "name": "Button",
      "request": undefined,
    },
  },
  "doc": Object {
    "component": "Button",
    "components": Object {
      "Button": "dc86c4457416c54c13147e780d972747",
    },
    "title": "Storybook/MDX",
  },
  "packages": Object {},
  "stories": Object {},
  "transformed": "
    
    import { Meta } from '@component-controls/blocks';
import { Btn as Button } from '../../components/button-named-class';
<Meta title=\\"Storybook/MDX\\" component={Button} />
",
}
`;

exports[`mdx-component propstable-of.mdx 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "title": "Story",
  },
  "packages": Object {},
  "stories": Object {},
  "transformed": "
    
    import { Meta, PropsTable } from '@component-controls/blocks';
import ArrowButton from '../../components/button-default-arrow-func';

<Meta title=\\"Story\\" />

<PropsTable of={ArrowButton} />

",
}
`;

exports[`mdx-component story-import.mdx 1`] = `
Object {
  "components": Object {
    "1e542c3ba23e8699f77ec19bf3f9e3ea": Object {
      "from": "./Button",
      "importedName": "Btn",
      "imports": undefined,
      "name": "Button",
      "request": undefined,
    },
    "7e2f74e50bd73c6a0bf50c40d3c149d4": Object {
      "from": "./Toggle",
      "importedName": "Toggle",
      "imports": undefined,
      "name": "Toggle",
      "request": undefined,
    },
  },
  "doc": Object {
    "component": "Button",
    "components": Object {
      "Button": "1e542c3ba23e8699f77ec19bf3f9e3ea",
      "Toggle": "7e2f74e50bd73c6a0bf50c40d3c149d4",
    },
    "title": "Storybook/MDX",
  },
  "packages": Object {},
  "stories": Object {
    "myStory": Object {
      "component": "Toggle",
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 50,
          "line": 23,
        },
        "start": Object {
          "column": 6,
          "line": 23,
        },
      },
      "name": "my-story",
      "source": "<Story name=\\"my-story\\" component={Toggle} />",
    },
  },
  "transformed": "
    
    import { Meta, Story } from '@component-controls/blocks';
import { Btn as Button } from './Button';
import { Toggle } from './Toggle';

<Meta title=\\"Storybook/MDX\\" component={Button} />

<Story name=\\"my-story\\" component={Toggle} />
",
}
`;

exports[`mdx-component story-subcomponents.mdx 1`] = `
Object {
  "components": Object {
    "2f4bd34d15be236674abaf6e894afe6e": Object {
      "from": "../../components/button-named-arrow-func",
      "importedName": "Button",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "Button",
      "request": "/Users/atanasster/component-controls/core/instrument/test/fixtures/components/button-named-arrow-func.js",
    },
    "dc262d21cd7820523c7e0bacca461e87": Object {
      "from": "../../components/button-default-arrow-func",
      "importedName": "default",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "ArrowButton",
      "request": "/Users/atanasster/component-controls/core/instrument/test/fixtures/components/button-default-arrow-func.js",
    },
  },
  "doc": Object {
    "components": Object {
      "ArrowButton": "dc262d21cd7820523c7e0bacca461e87",
      "Button": "2f4bd34d15be236674abaf6e894afe6e",
    },
    "title": "Story",
  },
  "packages": Object {},
  "stories": Object {
    "story": Object {
      "component": "ArrowButton",
      "id": "story",
      "loc": Object {
        "end": Object {
          "column": 8,
          "line": 29,
        },
        "start": Object {
          "column": 6,
          "line": 23,
        },
      },
      "name": "story",
      "source": "<Story
        name=\\"story\\"
        component={ArrowButton}
        subcomponents={{
          'My Button Tab': Button,
        }}
      />",
      "subcomponents": Object {
        "My Button Tab": "Button",
      },
    },
  },
  "transformed": "
    
    import { Meta, Story } from '@component-controls/blocks';
import ArrowButton from '../../components/button-default-arrow-func';
import { Button } from '../../components/button-named-arrow-func';

<Meta title=\\"Story\\" />

<Story name=\\"story\\" component={ArrowButton} subcomponents={{ 'My Button Tab': Button }} />

",
}
`;

exports[`mdx-component subcomponents.mdx 1`] = `
Object {
  "components": Object {
    "2f4bd34d15be236674abaf6e894afe6e": Object {
      "from": "../../components/button-named-arrow-func",
      "importedName": "Button",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "Button",
      "request": "/Users/atanasster/component-controls/core/instrument/test/fixtures/components/button-named-arrow-func.js",
    },
    "dc262d21cd7820523c7e0bacca461e87": Object {
      "from": "../../components/button-default-arrow-func",
      "importedName": "default",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "ArrowButton",
      "request": "/Users/atanasster/component-controls/core/instrument/test/fixtures/components/button-default-arrow-func.js",
    },
  },
  "doc": Object {
    "component": "ArrowButton",
    "components": Object {
      "ArrowButton": "dc262d21cd7820523c7e0bacca461e87",
      "Button": "2f4bd34d15be236674abaf6e894afe6e",
    },
    "subcomponents": Object {
      "Button": "Button",
    },
    "title": "Story",
  },
  "packages": Object {},
  "stories": Object {},
  "transformed": "
    
    import { Meta, Story } from '@component-controls/blocks';
import ArrowButton from '../../components/button-default-arrow-func';
import { Button } from '../../components/button-named-arrow-func';

<Meta title=\\"Story\\" component={ArrowButton} subcomponents={{ Button }} />

",
}
`;
